Sources/MapboxMaps/Foundation/MapboxMap.swift

```diff
@@ -511,6 +511,66 @@ extension MapboxMap: AttributionDataSource {
     }
 }
 
+// MARK: - Feature State -
+
+extension MapboxMap {
+
+    /// Update the state map of a feature within a style source.
+    /// Update entries in the state map of a given feature within a style source. Only entries listed in the state map
+    /// will be updated. An entry in the feature state map that is not listed in `state` will retain its previous value.
+    ///
+    /// - Note: updates to feature state are asynchronous, so changes made by this method migth
+    ///         not be immediately visible using getStateFeature().
+    ///
+    /// - Parameters:
+    ///   - sourceId: Style source identifier
+    ///   - sourceLayerId: Style source layer identifier (for multi-layer sources such as vector sources). Defaults to `nil`.
+    ///   - featureId: Identifier of the feature whose state should be updated
+    ///   - state: Map of entries to update with their respective new values
+    public func setFeatureState(sourceId: String, sourceLayerId: String? = nil, featureId: String, state: [String: Any]) {
+        __map.setFeatureStateForSourceId(sourceId,
+                                         sourceLayerId: sourceLayerId,
+                                         featureId: featureId,
+                                         state: state)
+    }
+
+    /// Get the state map of a feature within a style source.
+    ///
+    /// - Note: Updates to feature state are asynchronous, so changes made by other methods might not be immediately visible.
+    /// - Parameters:
+    ///   - sourceId: Style source identifier.
+    ///   - sourceLayerId: Style source layer identifier (for multi-layer sources such as vector sources).
+    ///   - featureId: Identifier of the feature whose state should be queried.
+    ///   - callback: Feature's state map or an empty map if the feature could not be found.
+    public func getFeatureState(sourceId: String, sourceLayerId: String? = nil, featureId: String, callback: @escaping (Result<[String: Any], Error>) -> Void) {
+        __map.getFeatureState(forSourceId: sourceId,
+                              sourceLayerId: sourceLayerId,
+                              featureId: featureId,
+                              callback: coreAPIClosureAdapter(for: callback,
+                                                              type: NSDictionary.self,
+                                                              concreteErrorType: MapError.self))
+    }
+
+    /// Removes entries from a feature state object.
+    /// Remove a specified property or all property from a feature's state object, depending on the value of `stateKey`.
+    ///
+    /// - Note: Updates to feature state are asynchronous, so changes made by this method
+    ///         might not be immediately visible using `getStateFeature`.
```

Comment:
Similar concerns here about async. Seems like it'd be nearly impossible for developers to write correct code if we can't make some guarantees around ordering.

Reply:
Red herring, updated documentation. These calls are synchronous.